lib/DebugInfo/DebugInfo.cpp

```diff
@@ -0,0 +1,35 @@
+/// \file DebugInfo.cpp
+/// \brief Run an instance of DebugHelper over the IR.
+
+#include "llvm/IR/AssemblyAnnotationWriter.h"
+#include "llvm/IR/IRBuilder.h"
+
+#include "revng/DebugInfo/DebugInfo.h"
+#include "revng/Support/CommandLine.h"
+#include "revng/Support/Debug.h"
+#include "revng/Support/DebugHelper.h"
+#include "revng/Support/revng.h"
+
+using namespace llvm;
+
+char DebugInfo::ID = 0;
+static RegisterPass<DebugInfo> X("debug-info", "Add debug metadata");
+
+cl::opt<std::string> DebugPath("debug-info-path",
+                               cl::desc("Filename produced as output"),
+                               cl::value_desc("filename"),
+                               cl::NumOccurrencesFlag::Optional);
+
+bool DebugInfo::runOnModule(Module &M) {
+  if (DebugPath.size() == 0) {
+    DebugHelper DBG("debug_info",
+                    &M,
+                    DebugInfoType::LLVMIR,
+                    M.getModuleIdentifier());
+    DBG.generateDebugInfo();
+  } else {
+    DebugHelper DBG("debug_info", &M, DebugInfoType::LLVMIR, DebugPath);
+    DBG.generateDebugInfo();
+  }
```

Comment:
* [ ] Use the `if` for the path only, the rest is identical.
* [ ] It makes me a bit sad that we have to specify a path. It'd be nice to be able to run `opt --debug-info` and just get the final output decorated as appropriately. However, I think is not possible due to the fact that debug info need to explicitly reference a path.
* [ ] Why not have a switch for the other `DebugInfoType`s?

Reply:
Improved the `if` in 7920dab4d3755c4e576b8de736184a8ce4ddac8e.

For the second point, yeah I'm sad too but unfortunately, as you pointed out, due to the fact that we are actually abusing a bit the `DILocation` mechanism the standard way to interact with `opt` is not adequate (e.g. the double serialization pass just to collect the line at which a certain instruction will be serialized in a `.ll` so that we can fix the correct `DILocation`, and therefore the fact that this pass does not make sense unless it is executed as the last one (we discussed more on this aspects in this old discussion on the old PR https://github.com/revng/revng/pull/37#discussion_r564695242).

For the third point, yes in general I guess we can have the switch, for the moment I just focused on decorating llvm ir beacuse I never used it for anything else. 
Practically, how would you takle the task? I see two main problems:
- The first, how to specify the `DebugInfoType` wanted? Another `cl::option`?
- The second, if we implement the switch we also need to take care that the name of the produced module is coherent with the argument (simplifying, `.ll` for LLVMIR, `.S` etc...). Do we just rely on the fact that the user specify coherent things across the `DebugInfoType` and extension, or we manually add the coherent extension ourselves?